### PR TITLE
METRON-1424: Kerberos: Solr

### PR DESCRIPTION
**Pull request by merrimanr**
## Contributor Comments
This PR adds Kerberos support for Solr in Metron.  This has been verified in full dev using the following steps:

1. Spin up full dev
2. Stop and remove Elasticsearch and Kibana
3. Install HDP Search (https://docs.hortonworks.com/HDPDocuments/HDP2/HDP-2.6.0/bk_solr-search-installation/content/ch_hdp-search-install-ambari.html)
4. Create collections for bro, snort and error using the `$METRON_HOME/bin/create_collection.sh` script.
5. Kerberize full dev using the instructions in https://github.com/apache/metron/blob/master/metron-deployment/Kerberos-manual-setup.md
6. In Ambari, change the "Solr Zookeeper Urls" setting in the Metron > Index Settings tab to "node1:2181/solr" and the "Random Access Search Engine" setting in the Metron > Indexing tab to "Solr" (it also helps to change "Random Access Indexing Offset" in the Metron > Indexing tab to "LATEST")
7. Verify data is showing up in Solr
8. Restart Metron REST and you should be able to query data in Solr without issue

This PR assumes HDP Search is being used.  The benefit of using HDP Search is that Ambari handles Kerberos configuration for Solr.  If using a separate Solr install, Kerberos configuration would need to be done manually there.

Side note:  I had a lot of trouble getting everything to work all at once due to resource constraints in full dev.  I would suggest shutting down as many services as possible, getting data into Solr first, then shutting down topologies, then starting REST and querying data.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [ ] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [ ] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [ ] Have you included steps or a guide to how the change may be verified and tested manually?
- [ ] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [ ] Have you written or updated unit tests and or integration tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: metron-platform/metron-solr/src/main/java/org/apache/metron/solr/writer/SolrWriter.java [hunk 1]
```diff
@@ -239,4 +245,12 @@ public void close() throws Exception {
       solr.close();
     }
   }
+
+  private boolean isKerberosEnabled(Map stormConfig) {
+    if (stormConfig == null) {
+      return false;
+    }
+    String value = (String) stormConfig.get("java.security.auth.login.config");
```
Comment (cestella) [inline metron-platform/metron-solr/src/main/java/org/apache/metron/solr/writer/SolrWriter.java]: Can we make this a constant?

Comment (merrimanr) [reply metron-platform/metron-solr/src/main/java/org/apache/metron/solr/writer/SolrWriter.java]: Done

**Comment by mmiklavc** [comment]
+1 by inspection. Nice work @merrimanr

**Comment by cestella** [comment]
+1

**Pull request closed without merging**